socketshark/utils.py

```diff
@@ -7,6 +7,24 @@
 from . import constants as c
 
 
+def _get_rate_limit_wait(resp, opts):
+    """
+    Returns the number of seconds we should wait given a 429 HTTP response and
+    HTTP options.
+    """
+    reset_header = opts['rate_limit_reset_header_name']
+    if reset_header and reset_header in resp.headers:
+        try:
+            wait = int(resp.headers[reset_header])
```

Comment:
This needs to be `float`.  When we switch to token buckets the sleeps will almost always be between 0 and 1.0.